### DISCUSSION TITLE
Who calls INotificationHandlers<xNotification>?

### DISCUSSION THREAD

**Discussion by 1304654** [kgrzybek/modular-monolith-with-ddd#246] [Q&A]
I would like to know how the notification handlers are executed. I was see the DomainEventsDispatcher that has:
```
 foreach (var domainEvent in domainEvents)
            {
                await _mediator.Publish(domainEvent);
            }
```
But no NotificationHandlers listens for IDomainEvent. 

All NotificationHandlers listen for DomainNotificationBase. So who throws DomainNotificationBases like MemberSubscriptionExpirationDateChangedNotification through the bus? For the corresponding NotificationHandler to be executed.

**Comment by musjj** [comment]
Not really familiar with C#, so I would like to know about this too.
